### PR TITLE
Adding "wpcom-elasticsearch" plugin

### PR DESCRIPTION
**Pull request by emrikol**
This plugin will help power the core WordPress search via Elasticsearch from WordPress.com.  It will not work without proper setup from WordPress.com VIP support.

### CODE REVIEW

File: shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php [hunk 1]
```diff
@@ -0,0 +1,821 @@
...
+		$es_query_args = wpcom_search_api_wp_to_es_args( $es_wp_query_args );
+
+		$es_query_args['fields'] = array(
+			'post_id',
+			'blog_id'
+		);
+
+		// Ask for a search suggestion (to catch typos)
+		// See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-suggesters-phrase.html for options
+		$suggest_beta_blogs = array(
```
Comment (ethitter) [inline shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php]: Is this necessary?

File: shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php [hunk 2]
```diff
@@ -0,0 +1,821 @@
...
+	 * This method accepts an array of additional blog names, in the format of vip.wordpress.com,
+	 * that should be queried.
+	 * 
+	 * @param array $domains The additional blogs to query
+	 */
+	public function set_additional_blogs( $domains = array() ) {
+		$this->additional_indices = array();
+
+		foreach( $domains as $domain ) {
+			$blog_id = (int) get_blog_id_from_url( $domain );
```
Comment (ethitter) [inline shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php]: This should be defined as a stub somewhere, at the least.

File: shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php (hunk 1 above)
Comment (joshbetz) [inline shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php]: We can probably delete this or enable the suggest feature for all sites on Go? cc @nickdaugherty

Comment (mjangda) [inline shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php]: Removed for now: 71cffcc

Comment (mjangda) [inline shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php]: Removed for now: 71cffcc

**Comment by mjangda** [comment]
We should probably have a check to see that Jetpack is enabled.

**Pull request merged**